include/analyzedphenotypes.admin.form.inc

```diff
@@ -1061,410 +1061,348 @@ function analyzedphenotypes_dragdrop($form, &$form_state, $source_name) {
+    // if a field has been selected. If a field has been altered
+    // check cv and databse field and ensure both are set.
+    foreach($vars as $type => $v) {
+      if ((int)$form_state['values'][$v] > 0) {
+        $var_ctr++;
+      }
+    }
 
-          $cvterm_options = array_map(function($t) {
-            return $t['name'];
-          }, $cvterms);
-        }
-        else {
-          $default_cvterm = 0;
-          $cvterm_options = array();
-        }
+    if ($var_ctr > 0) {
+      // Field set.
+      if ($form_state['values'][$vars['cv']] <= 0) {
+        // Controlled vocabulary.
+        form_set_error($vars['cv'], 'Please select Controlled Vocabulary.');
+      }
 
-        $form['ap_AJAX_wrapper_getcvterms' . $title][$var['def']] = array(
-          '#type' => 'select',
-          '#title' => 'Select default term',
-          '#options' => $cvterm_options,
-          '#default_value' => $default_cvterm,
-          '#description' => t('Select Controlled Vocabulary (CV) and CV Term.'),
-          '#id' => $i . '_term_select_field',
-        );
+      if ($form_state['values'][$vars['db']] <= 0) {
+        // Associated databse.
+        form_set_error($vars['db'], 'Please select Associated Database.');
       }
+    }
+  }
 
-      break;
+  // CHECK CONTROLLED VOCABULARY TERMS:
+  $terms = analyzedphenotypes_systemvariables('terms');
 
-      //
-      // Define additional case here.
+  $failed_ctr = 0;
 
-      //
-      default:
-        $base_admin = 'admin/tripal/extension/analyzedphenotypes';
-
-        $header = array('Section', 'Information', 'View');
-        $rows[] = array('Crop Ontology', 'Map a Genus to a Crop Ontology Vocabularies.', l('view', $base_admin . '/settings/crop_ontology'));
-        $rows[] = array('Database and CV Terms', 'Set the default Database and Controlled Vocabulary.', l('view', $base_admin . '/settings/database_cv'));
-        $rows[] = array('Plant Property Traits', 'Define Controlled Vocabulary for Plant Property Traits location, Replicate, Year and Data Collector.', l('view', $base_admin . '/settings/property_terms'));
-
-        $form['ap_table'] = array(
-          '#markup' => theme('table', array(
-            'header' => $header,
-            'rows' => $rows,
-            'attributes' => array(
-              'id' => 'ap-settings-info-table',
-              'sticky' => FALSE,
-            ),
-          )),
-        );
+  // Test if value entered (field for terms are autocomplete).
+  foreach($terms as $term => $t) {
+    $field_term = $form_state['values'][$t];
 
-        return $form;
-  }
+    if (empty($field_term)) {
+      // Field is empty. Mark field.
+      form_set_error($t, '');
+      $failed_ctr++;
+    }
+    else {
+      // Check if term in cvterm table.
+      $is_term = analyzedphenotypes_cvprop('term_cv', array(
+        'term_cv' => $field_term,
+      ));
 
+      if (is_null($is_term)) {
+        // Invalid term. Mark field.
+        form_set_error($t, '');
```

Comment:
See last comment. A good message would be "[term entered] is not a valid controlled vocabulary term. Please confirm that it already exists keeping in mind that case and whitespace are important."